### DISCUSSION TITLE
Help understanding side-effects of DID creation

### DISCUSSION THREAD

**Discussion by samrae7** [decentralized-identity/veramo#766] [Q&A] [answered]
I am following the veramo node tutorial [here](https://veramo.io/docs/node_tutorials/node_setup_identifiers) and I have a fundamental misunderstanding which I'd like some help with.

I followed the steps exactly so that I created an infura project and used its id. Then created a DID as outlined. I got the expected json printed to my console.

However, I thought that what was happening is that an identity (which I think represents a wallet address on ethereum - ie. a public key that represents an 'identity') was created on-chain (actually on  rinkeby test chain).

But if that was the case I would expect to see stats for this in the Infura dashboard for my project, or even find the public key for my DID here: [https://rinkeby.etherscan.io/ ](https://rinkeby.etherscan.io/), but I can't see either

Is this not happening because the did is created 'off-chain'? If so then I would really appreciate some conceptual help - surely when I create a DID using @veramo/did-provider-ethr something has to happen on-chain, otherwise what is the role of the blockchain at all?

**Comment by mirceanis** [comment, marked as answer]
Hi @samrae7,
What you are seeing is correct behavior.
`did:ethr` is designed to be free to create and not have on-chain activity for onboarding.
Creating an ID using `did:ethr` is equivalent to creating a new account in a wallet such as Metamask. There is no on-chain activity until you need to change something about the ID.

In the case of `did:ethr`, with Veramo, you can add or remove keys or services with the basic `@veramo/did-manager` API.
For these operations a blockchain transaction is needed, which will then be visible on etherscan.
This will require gas, though; so you will have to fund the owner address to be able to perform these operations.

You can read more about `did:ethr` in [the spec](https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md)

I hope this helps

**Comment by samrae7** [comment]
Thank you @mirceanis. This clears up a lot of confusion for me. I read the first part of the spec actually and saw that it is saying that there is no need for 'registration' but wasn't sure of the exact meaning of that as I am new to blockchain and ether.

One small follow-up question: in the case of using veramo to create a `credential` for a particular DID is that also off-chain? From my understanding so far the answer is no because it is a change that will be reflected in the DID document.

**Reply by mirceanis** [reply]
Actually no, credentials are not stored on-chain, and they should never be stored on-chain.
There is no way to delete them from the blockchain once they are stored there.

The DID document is not like a Resumé for a DID, but rather more like a pseudonymous business card.
It usually contains public keys that can be used to verify signatures or to establish shared secrets, and can also contain service endpoints, that can be used to interact with that DID in some form.

Credentials are created off-chain and then exchanged between an issuer and a holder or presented by the holder to a verifier.

I hope this clears some confusion, but please ask some more if there are still missing pieces to the puzzle.
This can seem overwhelming at first.

**Reply by samrae7** [reply]
Thank you @mirceanis for taking the time to answer, it's really helpful
